nat-lab/tests/test_lana.py

```diff
@@ -1710,15 +1685,14 @@ async def test_lana_with_disconnected_node(
             .add_received_data_validator(
                 exists=False,
             )
-            # TODO enable this when NAT traversal reporting is implemented (LLT-4858)
-            # .add_nat_traversal_conn_info_validator(
-            #     exists=False,
-            # )
             .add_derp_conn_info_validator(
                 exists=True,
                 servers=DERP_SERVERS_STRS,
             )
         )
+        add_nat_traversal_info_validator_to_non_symmetric_clients(
```

Comment:
Keeping up with the above architecture, couldn't we move the condition checking which public key if lower directly into `add_nat_traversal_conn_info_validator` ?

Reply:
Ummm ... I guess I can move that, and it would be logical in some sense, but the question is about the maintenance. This one clearly states, that this for `non_symmetrical_clients`, and someone later can add `add_nat_traversal_conn_info_validator` to all clients (including symmetrical ones) and have a headache after that :smile: .

Reply:
To combine them both, we could add an additional field of `client_type` in `add_nat_traversal_conn_info_validator` which can be `symmetric` or `asymmetric` and further decisions could be based on that. I feel it would be better to deal with both types in a homogeneous way.

Reply:
Fixed